gtest C++14 requirement (move to C++17)

Google Test since version 1.13.0 requires C++14:

```
[4/69] g++ -Itest/tagParsing.p -Iinclude -I../include -I/usr/include/p11-kit-1 -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c++11 -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -DGTEST_HAS_PTHREAD=1 -pthread -MD -MQ test/tagParsing.p/tagParsing.cpp.o -MF test/tagParsing.p/tagParsing.cpp.o.d -o test/tagParsing.p/tagParsing.cpp.o -c ../test/tagParsing.cpp
FAILED: test/tagParsing.p/tagParsing.cpp.o 
g++ -Itest/tagParsing.p -Iinclude -I../include -I/usr/include/p11-kit-1 -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c++11 -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -DGTEST_HAS_PTHREAD=1 -pthread -MD -MQ test/tagParsing.p/tagParsing.cpp.o -MF test/tagParsing.p/tagParsing.cpp.o.d -o test/tagParsing.p/tagParsing.cpp.o -c ../test/tagParsing.cpp
In file included from /usr/include/gtest/gtest-message.h:57,
                 from /usr/include/gtest/gtest-assertion-result.h:46,
                 from /usr/include/gtest/gtest.h:64,
                 from ../test/tagParsing.cpp:20:
/usr/include/gtest/internal/gtest-port.h:270:2: error: #error C++ versions less than C++14 are not supported.
  270 | #error C++ versions less than C++14 are not supported.
      |  ^~~~~
```

--- a/meson.build
+++ b/meson.build
@@ -1,7 +1,7 @@
 project('libkiwix', 'cpp',
   version : '12.1.0',
   license : 'GPLv3+',
-  default_options : ['c_std=c11', 'cpp_std=c++11', 'werror=true'])
+  default_options : ['c_std=c11', 'cpp_std=c++17', 'werror=true'])
 
 compiler = meson.get_compiler('cpp')
 

